src/provider/qti.js

```diff
@@ -495,15 +495,21 @@ var qtiProvider = {
 
                 this.setState('closedOrSuspended', true);
 
+                let params = {
+                    itemDefinition: testContext.itemIdentifier,
+                    reason: {
+                        reasons: data && data.reasons,
+                        comment: data && (data.originalMessage || data.message)
+                    }
+                };
+
+                const itemState = self.itemRunner.getState();
+                if (Object.keys(itemState).length) {
+                    params = Object.assign({}, params, { itemState });
```

Comment:
Now I guess we'll lose answers user changed this time, since `move` failed and his answer wasn't actually submitted. (You've written about `answerCache` in description). It could be ok (we did fix the main bug!), but if not, could we think of this as a continuation of failed `move` action and pass it `itemState` which was supposed to go into `move`? 

`pause` is called [at least] here
a. https://github.com/oat-sa/tao-test-runner-qti-fe/blob/develop/src/provider/qti.js#L714
b. https://github.com/oat-sa/tao-test-runner-qti-fe/blob/develop/src/provider/qti.js#L263 --> our case?
Then for 'b.' something like:
```
function computeNext(action, params, loadPromise) {
   self.trigger('pause', {message : err.message, moveParams: params});
}
...
.on('pause', function(data) {
	this.getProxy().callTestAction('pause', {
		itemState: moveParams && moveParams.itemState && moveParams.itemIdentifier === testContext.itemIdentifier ? moveParams.itemState : self.itemRunner.getState()
```

Note 1: I hope it's fine to have `itemState` without `itemResponses`...
Note 2: If we decide not to send user's latest changes in this case, could it be better not to send `itemState` at all? In case 'a.' itemRunner can be in corrupted state? Though maybe 'pause' is called from other places too and they need state to be preserved.

Reply:
I think we can add to case _b._  extra params, because it also will be called after clearing item runner and we will get the same situation: item state is cleared. My fix will not work, if "move" action is finished with error.

However we can not avoid sending `itemState` in "pause" handler. For case where user working with test item and in this moment he get "paused" state, storing current state is absolutely necessary, because after resuming we always get actual state from server and user will be very sad, if he loses his temporary state (especially if test - writing an essay).

Reply:
Update: for me `move` always completes with error in this case. But for you (and looks like Martin too) it succeeds. Then this discussion isn't relevant to the main case, so let's forget it.

![Screenshot_38](https://user-images.githubusercontent.com/38751932/146531199-45f7ca7e-e1cb-401d-83b0-adf08ced1555.png)